rust_dev_preview/concurrency/src/lib.rs

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use futures::future;
+use std::fmt::{self, Display, Formatter};
+use std::future::Future;
+use std::str::FromStr;
+use std::sync::Arc;
+use tokio::sync::Semaphore;
+use tokio::time::Instant;
+use tracing::{debug, info};
+
+#[derive(Clone, Debug)]
+pub enum Runtime {
+    SingleThreaded,
+    MultiThreaded,
+}
+
+impl Display for Runtime {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        write!(
+            f,
+            "{}",
+            match self {
+                Runtime::SingleThreaded => "single-threaded",
+                Runtime::MultiThreaded => "multi-threaded",
+            }
+        )
+    }
+}
+
+#[derive(Debug)]
+pub struct RuntimeParseErr;
+
+impl Display for RuntimeParseErr {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        write!(f, "couldn't parse the runtime argument")
+    }
+}
+
+impl std::error::Error for RuntimeParseErr {}
+
+impl FromStr for Runtime {
+    type Err = RuntimeParseErr;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let is_single_threaded = s.eq_ignore_ascii_case("single")
```

Comment:
This feels like overkill

Reply:
You're right. We can have Clap do this for us